oauth2_http/javatests/com/google/auth/oauth2/UserAuthorizerTest.java

```diff
@@ -219,10 +226,69 @@ public void getCredentials_nullUserId_throws() throws IOException {
   public void getCredentials_refreshedToken_stored() throws IOException {
     final String accessTokenValue1 = "1/MkSJoj1xsli0AccessToken_NKPY2";
     final String accessTokenValue2 = "2/MkSJoj1xsli0AccessToken_NKPY2";
-    AccessToken accessToken1 = new AccessToken(accessTokenValue1, new Date(EXPIRATION_TIME));
+    AccessToken accessToken1 =
+        AccessToken.newBuilder()
+            .setTokenValue(accessTokenValue1)
+            .setExpirationTime(new Date(EXPIRATION_TIME))
+            .setScopes(GRANTED_SCOPES)
+            .build();
     MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
     transportFactory.transport.addClient(CLIENT_ID_VALUE, CLIENT_SECRET);
-    transportFactory.transport.addRefreshToken(REFRESH_TOKEN, accessTokenValue2);
+    transportFactory.transport.addRefreshToken(REFRESH_TOKEN, accessTokenValue2, GRANTED_SCOPES);
+    TokenStore tokenStore = new MemoryTokensStorage();
+    UserAuthorizer authorizer =
+        UserAuthorizer.newBuilder()
+            .setClientId(CLIENT_ID)
+            .setScopes(SCOPES)
+            .setTokenStore(tokenStore)
+            .setHttpTransportFactory(transportFactory)
+            .build();
+
+    UserCredentials originalCredentials =
+        UserCredentials.newBuilder()
+            .setClientId(CLIENT_ID_VALUE)
+            .setClientSecret(CLIENT_SECRET)
+            .setRefreshToken(REFRESH_TOKEN)
+            .setAccessToken(accessToken1)
+            .setHttpTransportFactory(transportFactory)
+            .build();
+
+    authorizer.storeCredentials(USER_ID, originalCredentials);
+
+    UserCredentials credentials1 = authorizer.getCredentials(USER_ID);
+
+    assertEquals(REFRESH_TOKEN, credentials1.getRefreshToken());
+    assertEquals(accessTokenValue1, credentials1.getAccessToken().getTokenValue());
```

Comment:
check there were no scopes before?

Reply:
There would be scopes before as well. They come as part of both first token response and future refresh responses as well.